### PR TITLE
home-assistant: support ista_ecotrend component

### PR DESCRIPTION
**Pull request by oynqr**
## Description of changes

This adds all relevant packages for supporting the [ista EcoTrend](https://www.home-assistant.io/integrations/ista_ecotrend/) Home Assistant integration. Since this is my first contribution, I have also added myself as a maintainer.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pyjson/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyjson/default.nix:15]: Use `fetchFromGitHub` and `sourceRoot` instead. That way we get a fetcher that supports automatic updates.

File: pkgs/development/python-modules/pyjson/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyjson/default.nix:21]:
```suggestion
  ];

  doCheck = false; # no tests

  pythonImportsCheck = [ "pyjson" ];
```

File: pkgs/development/python-modules/pyecotrend-ista/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyecotrend-ista/default.nix:27]:
```suggestion
    hash = "sha256-NAJ3/37dlwOiP1cilxfADUcuUt1VKF+eo/3JKgSUo3M=";
```

File: pkgs/development/python-modules/pyecotrend-ista/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyecotrend-ista/default.nix:41]: We don't care about test coverage. Remove the dependency and patch it out

```nix
  src = {
    ...
  };

  postPatch = ''
    sed -i "/addopts =/d" pyproject.toml
  '';
```

https://github.com/Ludy87/pyecotrend-ista/blob/main/pyproject.toml#L24

File: pkgs/development/python-modules/pytest-dataset/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pytest-dataset/default.nix:37]:
```suggestion
  ];

  doCheck = false; # no tests

  pythonImportsCheck = [ "pytest_dataset" ];
```

**Review by mweinelt — CHANGES_REQUESTED** [review]
Please reformat all new python-packages with nixfmt-rfc-style.

File: pkgs/development/python-modules/pytest-dataset/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-dataset/default.nix:23]:
```suggestion
      --replace-fail "setuptools_scm ~= 6.0" "setuptools_scm >= 6.0"
```

File: pkgs/development/python-modules/pytest-dataset/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-dataset/default.nix:27]:
```suggestion
```
redundant

**Review by dotlambda — CHANGES_REQUESTED** [review]
The commit messages should be e.g. `python312Packages.pyjson: init at 1.4.5`.

File: pkgs/development/python-modules/pyjson/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyjson/default.nix:31]: could be
```suggestion
    homepage = "https://github.com/leeyoshinari/Small_Tool/tree/master/pyjson";
```
but that's optional.

File: pkgs/development/python-modules/pytest-dataset/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-dataset/default.nix:16]: If you want you can ask upstream to tag their releases.

**Review by dotlambda — APPROVED** [review]

File: pkgs/development/python-modules/pytest-dataset/default.nix
Comment (oynqr) [reply pkgs/development/python-modules/pytest-dataset/default.nix:16]: Currently looking into how hard it would be to get `pyecotrend-ista` to use `pytest-datadir` instead of  `pytest-dataset`.

**Comment by oynqr** [comment]
I opened a [PR](https://github.com/Ludy87/pyecotrend-ista/pull/197) upstream to remove the dependency on `pytest-dataset`. Let's wait until that is merged or closed.

**Comment by oynqr** [comment]
Alright, the PR is through and upstream has cut a new release, completely removing the dependency.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**
